src/objects/zcl_abapgit_objects.clas.abap

```diff
@@ -495,28 +496,43 @@ CLASS ZCL_ABAPGIT_OBJECTS IMPLEMENTATION.
         check_objects_locked( iv_language = zif_abapgit_definitions=>c_english
                               it_items    = lt_items ).
 
-        LOOP AT lt_tadir ASSIGNING <ls_tadir>.
-          li_progress->show( iv_current = sy-tabix
-                             iv_text    = |Delete { <ls_tadir>-obj_name }| ).
-
-          CLEAR ls_item.
-          ls_item-obj_type = <ls_tadir>-object.
-          ls_item-obj_name = <ls_tadir>-obj_name.
-          delete_obj(
-            iv_package = <ls_tadir>-devclass
-            is_item    = ls_item ).
-
-* make sure to save object deletions
-          COMMIT WORK.
-        ENDLOOP.
-
       CATCH zcx_abapgit_exception INTO lx_error.
         zcl_abapgit_default_transport=>get_instance( )->reset( ).
         RAISE EXCEPTION lx_error.
     ENDTRY.
 
+    DO 3 TIMES.
+      LOOP AT lt_tadir ASSIGNING <ls_tadir>.
+        lv_count = lv_count + 1.
+        li_progress->show( iv_current = lv_count
```

Comment:
line 492: `li_progress = zcl_abapgit_progress=>get_instance( lines( lt_tadir ) ).`

perhaps move the progress bar initialization to after the DO and before LOOP

Reply:
when an object is successfully deleted, the `lt_tadir` entry is removed. `lv_count` should therefore count the entries properly. The progress looked ok during tests.

Reply:
worst case scenario, everything fails during deletion, then lv_count would be 3x the size of `lines( lt_tadir )` ?

I think moving the progress indicator to line 520 will make it work